mesonbuild/compilers/asm.py

```diff
@@ -259,6 +260,34 @@ def depfile_for_object(self, objfile: str) -> T.Optional[str]:
         return None
 
 
+# https://downloads.ti.com/docs/esd/SPRUI04/
+class TILinearAsmCompiler(TICompiler, Compiler):
+    language = 'linearasm'
+
+    def __init__(self, ccache: T.List[str], exelist: T.List[str], version: str,
+                 for_machine: 'MachineChoice', info: 'MachineInfo',
+                 linker: T.Optional['DynamicLinker'] = None,
```

Comment:
Avoid quoting type annotations in newly introduced code. It is an older style and unnecessary since we now require a minimum python >=3.7 and started using `from __future__ import annotations`.